openml/flows/flow.py

```diff
@@ -1,80 +1,358 @@
 from collections import OrderedDict
+import re
+
+import six
 import xmltodict
-import sklearn
 
 from .._api_calls import _perform_api_call
-from .functions import _check_flow_exists
+from openml.util import oml_cusual_string
 
 
 class OpenMLFlow(object):
     """OpenML Flow. Stores machine learning models.
 
+    Flows should not be generated manually, but by the function
+    :meth:`openml.flows.create_flow_from_model`. Using this helper function
+    ensures that all relevant fields are filled in.
+
     Parameters
     ----------
-    model : scikit-learn compatible model
-        The model the flow consists of. The model needs to have fit and predict methods.
-    description : string
+    name : str
+        Name of the flow. Is used together with the attribute `external_version`
+        as a unique identifier of the flow.
+    description : str
         Description of the flow (free text).
-    creator : string
-        FIXME
-    contributor : string
-        FIXME
-    tag : string
-        FIXME
+    model : object
+        ML model which is described by this flow.
+    components : OrderedDict
+        Mapping from component identifier to an OpenMLFlow object.
```

Comment:
Yes, I will add further documentation. Basically, a component is a building block used by more than one flow, for example the description of a decision tree. It can be us inside adaboost, bagging, random forests, extremely randomized trees or on their own.